### PR TITLE
Feature error logging

### PR DESCRIPTION
**Pull request by everHannes**
adds more send / recv error logging

### CODE REVIEW

**Comment by everHannes** [comment]
@a-krebs not sure on the error returns I added but I thought when a call is meant to send something it should return when it failed
have a look whether that's fine to merge

File: src/meta.c [hunk 1]
```diff
@@ -64,9 +64,14 @@ int send_meta(meshlink_handle_t *mesh, connection_t *c, const char *buffer, int
 }
 
 void broadcast_meta(meshlink_handle_t *mesh, connection_t *from, const char *buffer, int length) {
-	for list_each(connection_t, c, mesh->connections)
-		if(c != from && c->status.active)
-			send_meta(mesh, c, buffer, length);
+	for list_each(connection_t, c, mesh->connections) {
+		if(c != from && c->status.active) {
+			int err = send_meta(mesh, c, buffer, length);
```
Comment (a-krebs) [inline src/meta.c:69]: whitespace

File: src/net_packet.c [hunk 2]
```diff
@@ -263,10 +263,13 @@ static void receive_packet(meshlink_handle_t *mesh, node_t *n, vpn_packet_t *pac
     if (n->status.blacklisted) {
         logger(mesh, MESHLINK_WARNING, "Dropping packet from blacklisted node %s", n->name);
     } else {
-	n->in_packets++;
-	n->in_bytes += packet->len;
+		n->in_packets++;
```
Comment (a-krebs) [inline src/net_packet.c:266]: whitespace

File: src/net_packet.c [hunk 3]
```diff
@@ -574,15 +578,24 @@ int send_packet(meshlink_handle_t *mesh, node_t *n, vpn_packet_t *packet) {
 
 void broadcast_packet(meshlink_handle_t *mesh, const node_t *from, vpn_packet_t *packet) {
 	// Always give ourself a copy of the packet.
-	if(from != mesh->self)
-		send_packet(mesh, mesh->self, packet);
+	if(from != mesh->self) {
+		int err = send_packet(mesh, mesh->self, packet);
+	    if(err) {
```
Comment (a-krebs) [inline src/net_packet.c:583]: whitespace

File: src/net_packet.c [hunk 4]
```diff
@@ -574,15 +578,24 @@ int send_packet(meshlink_handle_t *mesh, node_t *n, vpn_packet_t *packet) {
...
 
 	logger(mesh, MESHLINK_INFO, "Broadcasting packet of %d bytes from %s (%s)",
 			   packet->len, from->name, from->hostname);
 
-	for list_each(connection_t, c, mesh->connections)
-		if(c->status.active && c->status.mst && c != from->nexthop->connection)
-			send_packet(mesh, c->node, packet);
+	for list_each(connection_t, c, mesh->connections) {
+		if(c->status.active && c->status.mst && c != from->nexthop->connection) {
+			int err = send_packet(mesh, c->node, packet);
```
Comment (a-krebs) [inline src/net_packet.c:593]: whitespace

File: src/protocol_auth.c [hunk 5]
```diff
@@ -48,10 +48,13 @@ static bool send_proxyrequest(meshlink_handle_t *mesh, connection_t *c) {
 			char *port;
 
 			sockaddr2str(&c->address, &host, &port);
-			send_request(mesh, c, "CONNECT %s:%s HTTP/1.1\r\n\r", host, port);
+			int err = send_request(mesh, c, "CONNECT %s:%s HTTP/1.1\r\n\r", host, port);
 			free(host);
 			free(port);
-			return true;
+		    if(err) {
```
Comment (a-krebs) [inline src/protocol_auth.c:54]: whitespace

File: src/protocol_auth.c [hunk 6]
```diff
@@ -67,7 +70,11 @@ static bool send_proxyrequest(meshlink_handle_t *mesh, connection_t *c) {
 				memcpy(s4req + 8, mesh->proxyuser, strlen(mesh->proxyuser));
 			s4req[sizeof s4req - 1] = 0;
 			c->tcplen = 8;
-			return !send_meta(mesh, c, s4req, sizeof s4req);
+			int err = send_meta(mesh, c, s4req, sizeof s4req);
+		    if(err) {
```
Comment (a-krebs) [inline src/protocol_auth.c:74]: whitespace

File: src/protocol_auth.c [hunk 7]
```diff
@@ -116,7 +123,11 @@ static bool send_proxyrequest(meshlink_handle_t *mesh, connection_t *c) {
 				logger(mesh, MESHLINK_ERROR, "Error: send_proxyrequest i > len");
 				abort();
 			}
-			return !send_meta(mesh, c, s5req, sizeof s5req);
+			int err = send_meta(mesh, c, s5req, sizeof s5req);
+		    if(err) {
```
Comment (a-krebs) [inline src/protocol_auth.c:127]: whitespace

File: src/protocol_auth.c [hunk 8]
```diff
@@ -137,7 +148,11 @@ bool send_id(meshlink_handle_t *mesh, connection_t *c) {
 		if(!send_proxyrequest(mesh, c))
 			return false;
 
-	return !send_request(mesh, c, "%d %s %d.%d", ID, mesh->self->connection->name, mesh->self->connection->protocol_major, minor);
+	int err = send_request(mesh, c, "%d %s %d.%d", ID, mesh->self->connection->name, mesh->self->connection->protocol_major, minor);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_auth.c:152]: whitespace

File: src/protocol_auth.c [hunk 9]
```diff
@@ -167,7 +182,11 @@ static bool finalize_invitation(meshlink_handle_t *mesh, connection_t *c, const
 
 	//TODO: callback to application to inform of an accepted invitation
 
-	sptps_send_record(&c->sptps, 2, data, 0);
+	int err = sptps_send_record(&c->sptps, 2, data, 0);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_auth.c:186]: whitespace

File: src/protocol_auth.c [hunk 10]
```diff
@@ -282,10 +314,15 @@ bool id_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
...
+			logger(mesh, MESHLINK_ERROR, "Failed to retrieve invitation key\n");
 			return false;
-		if(0 != send_request(mesh, c, "%d %s", ACK, mykey))
+		}
+		int err = send_request(mesh, c, "%d %s", ACK, mykey);
+		if(err) {
+			logger(mesh, MESHLINK_ERROR, "id_h send_request failed with %d\n", err);
 			return false;
+		}
 		free(mykey);
```
Comment (a-krebs) [inline src/protocol_auth.c]: Memory leak, in the old code too. `free(mykey)` before returning.

File: src/protocol_edge.c [hunk 11]
```diff
@@ -173,8 +175,11 @@ bool add_edge_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 }
 
 bool send_del_edge(meshlink_handle_t *mesh, connection_t *c, const edge_t *e) {
-	return !send_request(mesh, c, "%d %x %s %s", DEL_EDGE, rand(),
-						e->from->name, e->to->name);
+	int err = send_request(mesh, c, "%d %x %s %s", DEL_EDGE, rand(), e->from->name, e->to->name);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_edge.c:179]: whitespace

File: src/protocol_key.c [hunk 12]
```diff
@@ -32,7 +32,10 @@
 #include "xalloc.h"
 
 void send_key_changed(meshlink_handle_t *mesh) {
-	send_request(mesh, mesh->everyone, "%d %x %s", KEY_CHANGED, rand(), mesh->self->name);
+	int err = send_request(mesh, mesh->everyone, "%d %x %s", KEY_CHANGED, rand(), mesh->self->name);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:36]: whitespace

File: src/protocol_key.c [hunk 13]
```diff
@@ -75,7 +78,11 @@ static int send_initial_sptps_data(void *handle, uint8_t type, const void *data,
 	to->sptps.send_data = send_sptps_data;
 	char buf[len * 4 / 3 + 5];
 	b64encode(data, buf, len);
-	return send_request(mesh, to->nexthop->connection, "%d %s %s %d %s", REQ_KEY, mesh->self->name, to->name, REQ_KEY, buf);
+	int err = send_request(mesh, to->nexthop->connection, "%d %s %s %d %s", REQ_KEY, mesh->self->name, to->name, REQ_KEY, buf);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:82]: whitespace

File: src/protocol_key.c [hunk 14]
```diff
@@ -84,8 +91,11 @@ bool send_req_key(meshlink_handle_t *mesh, node_t *to) {
 			return false;
 		}
 		logger(mesh, MESHLINK_DEBUG, "No ECDSA key known for %s (%s)", to->name, to->hostname);
-		send_request(mesh, to->nexthop->connection, "%d %s %s %d", REQ_KEY, mesh->self->name, to->name, REQ_PUBKEY);
-		return true;
+		int err = send_request(mesh, to->nexthop->connection, "%d %s %s %d", REQ_KEY, mesh->self->name, to->name, REQ_PUBKEY);
+	    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:95]: whitespace

File: src/protocol_auth.c (hunk 6 above)
Comment (everHannes) [reply src/protocol_auth.c:74]: comeon, you don't need to comment every single whitespace...

File: src/protocol_key.c [hunk 15]
```diff
@@ -107,9 +117,12 @@ static bool req_key_ext_h(meshlink_handle_t *mesh, connection_t *c, const char *
 	switch(reqno) {
 		case REQ_PUBKEY: {
 			char *pubkey = ecdsa_get_base64_public_key(mesh->self->connection->ecdsa);
-			send_request(mesh, from->nexthop->connection, "%d %s %s %d %s", REQ_KEY, mesh->self->name, from->name, ANS_PUBKEY, pubkey);
+			int err = send_request(mesh, from->nexthop->connection, "%d %s %s %d %s", REQ_KEY, mesh->self->name, from->name, ANS_PUBKEY, pubkey);
+		    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:121]: whitespace

File: src/protocol_key.c [hunk 16]
```diff
@@ -132,8 +145,11 @@ static bool req_key_ext_h(meshlink_handle_t *mesh, connection_t *c, const char *
 		case REQ_KEY: {
 			if(!node_read_ecdsa_public_key(mesh, from)) {
 				logger(mesh, MESHLINK_DEBUG, "No ECDSA key known for %s (%s)", from->name, from->hostname);
-				send_request(mesh, from->nexthop->connection, "%d %s %s %d", REQ_KEY, mesh->self->name, from->name, REQ_PUBKEY);
-				return true;
+				int err = send_request(mesh, from->nexthop->connection, "%d %s %s %d", REQ_KEY, mesh->self->name, from->name, REQ_PUBKEY);
+			    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:149]: whitespace

File: src/protocol_key.c [hunk 17]
```diff
@@ -234,7 +250,11 @@ bool req_key_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 			return true;
 		}
 
-		send_request(mesh, to->nexthop->connection, "%s", request);
+		int err = send_request(mesh, to->nexthop->connection, "%s", request);
+	    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:254]: whitespace

File: src/protocol_key.c [hunk 18]
```diff
@@ -296,13 +316,20 @@ bool ans_key_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 			char *address, *port;
 			logger(mesh, MESHLINK_DEBUG, "Appending reflexive UDP address to ANS_KEY from %s to %s", from->name, to->name);
 			sockaddr2str(&from->address, &address, &port);
-			send_request(mesh, to->nexthop->connection, "%s %s %s", request, address, port);
+			int err = send_request(mesh, to->nexthop->connection, "%s %s %s", request, address, port);
+		    if(err) {
```
Comment (a-krebs) [inline src/protocol_key.c:320]: whitespace

File: src/protocol_misc.c [hunk 19]
```diff
@@ -37,7 +37,11 @@ bool send_status(meshlink_handle_t *mesh, connection_t *c, int statusno, const c
 	if(!statusstring)
 		statusstring = "Status";
 
-	return !send_request(mesh, c, "%d %d %s", STATUS, statusno, statusstring);
+	int err = send_request(mesh, c, "%d %d %s", STATUS, statusno, statusstring);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_misc.c:41]: whitespace

File: src/protocol_misc.c [hunk 20]
```diff
@@ -56,11 +60,15 @@ bool status_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 	return true;
 }
 
-bool send_error(meshlink_handle_t *mesh, connection_t *c, int err, const char *errstring) {
+bool send_error(meshlink_handle_t *mesh, connection_t *c, int error, const char *errstring) {
 	if(!errstring)
 		errstring = "Error";
 
-	return !send_request(mesh, c, "%d %d %s", ERROR, err, errstring);
+	int err = send_request(mesh, c, "%d %d %s", ERROR, error, errstring);
```
Comment (a-krebs) [inline src/protocol_misc.c:67]: whitespace

File: src/protocol_misc.c [hunk 21]
```diff
@@ -80,7 +88,11 @@ bool error_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 }
 
 bool send_termreq(meshlink_handle_t *mesh, connection_t *c) {
-	return !send_request(mesh, c, "%d", TERMREQ);
+	int err = send_request(mesh, c, "%d", TERMREQ);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_misc.c:92]: whitespace

File: src/protocol_misc.c [hunk 22]
```diff
@@ -91,15 +103,23 @@ bool send_ping(meshlink_handle_t *mesh, connection_t *c) {
 	c->status.pinged = true;
 	c->last_ping_time = mesh->loop.now.tv_sec;
 
-	return !send_request(mesh, c, "%d", PING);
+	int err = send_request(mesh, c, "%d", PING);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_misc.c:107]: whitespace

File: src/protocol_misc.c [hunk 23]
```diff
@@ -91,15 +103,23 @@ bool send_ping(meshlink_handle_t *mesh, connection_t *c) {
...
 }
 
 bool ping_h(meshlink_handle_t *mesh, connection_t *c, const char *request) {
 	return send_pong(mesh, c);
 }
 
 bool send_pong(meshlink_handle_t *mesh, connection_t *c) {
-	return !send_request(mesh, c, "%d", PONG);
+	int err = send_request(mesh, c, "%d", PONG);
+    if(err) {
```
Comment (a-krebs) [inline src/protocol_misc.c:119]: whitespace

**Comment by a-krebs** [comment]
Will do whitespace at a later time. Going to merge.

**Pull request merged**